packages/react-form-state/src/FormState.ts

```diff
@@ -70,16 +72,31 @@ export default class FormState<
 
   static getDerivedStateFromProps<T>(newProps: Props<T>, oldState?: State<T>) {
     const newInitialValues = newProps.initialValues;
+    const newExternalErrors = newProps.externalErrors || [];
 
     if (oldState == null) {
-      return createFormState(newInitialValues);
+      return createFormState(newInitialValues, newExternalErrors);
     }
 
     const oldInitialValues = initialValuesFromFields(oldState.fields);
     const shouldReinitialize = !isEqual(oldInitialValues, newInitialValues);
+    const oldExternalErrors = oldState.externalErrors;
 
     if (shouldReinitialize) {
-      return createFormState(newInitialValues);
+      return createFormState(newInitialValues, newExternalErrors);
+    }
+
+    if (
+      newExternalErrors !== oldExternalErrors ||
+      newExternalErrors.length !== oldExternalErrors.length
+    ) {
+      return {
+        externalErrors: newExternalErrors,
+        fields: fieldsWithErrors(oldState.fields, [
```

Comment:
We can derive `errors` here by merging external errors  and remote errors.

Reply:
I think we can simplify the logic by not associating field with errors using getDerivedStateFromProps.
Instead, we should be compute field with errors in render function with memoization helper.
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization

Advantage:
- simpler logic
- createFormState, updateRemoteErrors, getDerivedStateFromProps don't need to associate field with external error